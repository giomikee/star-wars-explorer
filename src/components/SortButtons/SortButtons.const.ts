export const SORT_OPTIONS = {
  date: 'date',
  name: 'name',
} as const

export const SORT_BUTTONS_EVENTS = {
  date: 'sorted:date',
  name: 'sorted:name',
} as const
