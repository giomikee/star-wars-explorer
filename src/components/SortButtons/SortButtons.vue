<template>
  <v-container>
    <v-skeleton-loader v-if="isLoading" type="actions" />
    <div v-else class="d-flex justify-end ga-2">
      <span class="align-self-center">Sort by:</span>
      <v-btn-toggle
        v-model="selectedSort"
        divided
        mandatory
        variant="outlined"
      >
        <v-btn
          :append-icon="getAppendIcon(SORT_OPTIONS.name)"
          aria-label="Sort by name ascending/descending"
          :value="SORT_OPTIONS.name"
          @click="emitSelectedSort"
        >
          Name
        </v-btn>

        <v-btn
          :append-icon="getAppendIcon(SORT_OPTIONS.date)"
          aria-label="Sort by date ascending/descending"
          :value="SORT_OPTIONS.date"
          @click="emitSelectedSort"
        >
          Date
        </v-btn>

      </v-btn-toggle>
    </div>
  </v-container>
</template>

<script lang="ts" setup>
  import { useSortButtons } from './composables/useSortButtons'
  import { SORT_BUTTONS_EVENTS, SORT_OPTIONS } from './SortButtons.const'

  interface Props {
    isLoading: boolean
  }

  defineProps<Props>()

  const emit = defineEmits(Object.values(SORT_BUTTONS_EVENTS))

  const { selectedSort, getAppendIcon, emitSelectedSort } = useSortButtons(emit)
</script>

<script lang="ts">
  export default { name: 'SortButtons' }
</script>
