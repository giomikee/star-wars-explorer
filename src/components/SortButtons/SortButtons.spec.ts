import { shallowMount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import SortButtons from './SortButtons.vue'

const createWrapper = (isLoading: boolean) => shallowMount(SortButtons, {
  props: { isLoading },
  global: {
    stubs: {
      VBtnToggle: false,
      VBtnGroup: false,
    },
  },
})

describe('Given SortButtons component', () => {
  describe.each([true, false])('When component is rendered', isLoading => {
    describe(`And isLoading ${isLoading}`, () => {
      it('Then it should match the snapshot', () => {
        const { element } = createWrapper(isLoading)

        expect(element).toMatchSnapshot()
      })
    })
  })
})
