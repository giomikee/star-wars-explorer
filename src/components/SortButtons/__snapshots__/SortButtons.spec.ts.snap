// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given SortButtons component > When component is rendered > And isLoading false > Then it should match the snapshot 1`] = `
<div
  class="v-container v-locale--is-ltr"
>
  <div
    class="d-flex justify-end ga-2"
  >
    <span
      class="align-self-center"
    >
      Sort by:
    </span>
    <div
      class="v-btn-group v-btn-group--divided v-theme--dark v-btn-group--density-default v-btn-toggle"
    >
      
      <v-btn-stub
        aria-label="Sort by name ascending/descending"
        block="false"
        border="false"
        density="default"
        disabled="false"
        exact="false"
        flat="false"
        icon="false"
        loading="false"
        replace="false"
        ripple="true"
        size="default"
        slim="false"
        stacked="false"
        symbol="Symbol(vuetify:v-btn-toggle)"
        tag="button"
        tile="false"
        value="name"
        variant="elevated"
      />
      <v-btn-stub
        aria-label="Sort by date ascending/descending"
        block="false"
        border="false"
        density="default"
        disabled="false"
        exact="false"
        flat="false"
        icon="false"
        loading="false"
        replace="false"
        ripple="true"
        size="default"
        slim="false"
        stacked="false"
        symbol="Symbol(vuetify:v-btn-toggle)"
        tag="button"
        tile="false"
        value="date"
        variant="elevated"
      />
      
    </div>
  </div>
</div>
`;

exports[`Given SortButtons component > When component is rendered > And isLoading true > Then it should match the snapshot 1`] = `
<div
  class="v-container v-locale--is-ltr"
>
  <v-skeleton-loader-stub
    boilerplate="false"
    loading="false"
    loadingtext="$vuetify.loading"
    type="actions"
  />
</div>
`;
